# 1. Importing Libraries

In [1]:
# import libraries 
import pandas as pd
import numpy as np
import os

# 2. Importing Dataset - AB_US_2023.csv

In [2]:
path = r'/Users/bazillaimran/Documents/03-2024 Airbnb US Analysis'
path

'/Users/bazillaimran/Documents/03-2024 Airbnb US Analysis'

In [3]:
df = pd.read_csv(os.path.join(path,'02 Data','Original Data', 'AB_US_2023.csv'), index_col = False)

/var/folders/17/9z1yp8ss3r174mlj49ql3qgr0000gn/T/ipykernel_23368/3289265353.py:1: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(path,'02 Data','Original Data', 'AB_US_2023.csv'), index_col = False)


# 3. Descriptive Analysis

In [4]:
df.head()

id                                               name  host_id  \
0   958              Bright, Modern Garden Unit - 1BR/1BTH     1169   
1  5858                                 Creative Sanctuary     8904   
2  8142  Friendly Room Apt. Style -UCSF/USF - San Franc...    21994   
3  8339                    Historic Alamo Square Victorian    24215   
4  8739                Mission Sunshine, with Private Bath     7149   

          host_name neighbourhood_group     neighbourhood  latitude  \
0             Holly                 NaN  Western Addition  37.77028   
1  Philip And Tania                 NaN    Bernal Heights  37.74474   
2             Aaron                 NaN    Haight Ashbury  37.76555   
3              Rosy                 NaN  Western Addition  37.77564   
4      Ivan & Wendy                 NaN           Mission  37.76030   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0 -122.43317  Entire home/apt    202               2                383   
1 -122.42089  Entire home/apt    235              30                111   
2 -122.45213     Private room     56              32                  9   
3 -122.43642  Entire home/apt    575               9                 28   
4 -122.42197     Private room    110               1                770   

  last_review  reviews_per_month  calculated_host_listings_count  \
0     2/19/23               2.31                               1   
1      8/6/17               0.66                               1   
2    10/27/22               0.09                              13   
3     6/28/19               0.17                               2   
4     2/25/23               4.65                               2   

   availability_365  number_of_reviews_ltm           city  
0               128                     59  San Francisco  
1               365                      0  San Francisco  
2               365                      1  San Francisco  
3               365                      0  San Francisco  
4               159                     34  San Francisco

In [5]:
df.shape

(232147, 18)

In [6]:
df.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365', 'number_of_reviews_ltm', 'city'],
      dtype='object')

In [7]:
df.describe()

id       host_id       latitude      longitude  \
count  2.321470e+05  2.321470e+05  232147.000000  232147.000000   
mean   2.584580e+17  1.582248e+08      36.610585     -98.301436   
std    3.465985e+17  1.587164e+08       5.126523      19.706929   
min    6.000000e+00  2.300000e+01      25.957323    -123.089130   
25%    2.638896e+07  2.299242e+07      33.976225    -118.315111   
50%    4.896307e+07  1.005783e+08      36.190556     -97.727670   
75%    6.633014e+17  2.686930e+08      40.717440     -77.026222   
max    8.581014e+17  5.069384e+08      47.734010     -70.996000   

               price  minimum_nights  number_of_reviews  reviews_per_month  \
count  232147.000000   232147.000000      232147.000000      183062.000000   
mean      259.468001       13.495867          40.915230           1.638348   
std      1024.645918       27.920631          80.649152           1.910812   
min         0.000000        1.000000           0.000000           0.010000   
25%        91.000000        2.000000           1.000000           0.310000   
50%       149.000000        3.000000           9.000000           1.000000   
75%       250.000000       30.000000          43.000000           2.420000   
max    100000.000000     1250.000000        3091.000000         101.420000   

       calculated_host_listings_count  availability_365  number_of_reviews_ltm  
count                   232147.000000     232147.000000          232147.000000  
mean                        29.879055        180.985686              11.689701  
std                        106.013665        134.715299              20.599954  
min                          1.000000          0.000000               0.000000  
25%                          1.000000         52.000000               0.000000  
50%                          2.000000        175.000000               3.000000  
75%                         10.000000        321.000000              16.000000  
max                       1003.000000        365.000000            1314.000000

All columns except for "reviews_per_month" contain 232,147 entries, while the "reviews_per_month" column has 183,062 entries. This discrepancy suggests that there are missing values in the "reviews_per_month" column. Regarding the "price" column, the mean of $259.5 exceeds the median of $149.0, indicating a potential presence of outliers or very high price values, such as the maximum value of $100,000. Given that the median price is $149 and the 75th percentile is $250, a price of $100,000 is substantially higher than the majority of other prices per night in the dataset, suggesting potential outliers. Similarly, the "minimum_nights" column exhibits outliers, with a maximum count of 1250 or over 3 years, while the 75th percentile is 30, the median is 3, and the mean is 13.5. However, it's reasonable to assume that some customers may rent properties for longer durations than the typical stay.

In [8]:
df.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                               int64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
number_of_reviews_ltm               int64
city                               object
dtype: object

# 4. Data Consistency Checks

In [9]:
# Checking for Mixed-type columns
for col in df.columns.tolist():
  weird = (df[[col]].map(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df[weird]) > 0:
    print (col)

name
neighbourhood_group
last_review


In [10]:
# Handing mixed data types. 
# Covert name column mix type into string.
df['name'] = df['name'].astype('str')

In [11]:
# Covert host_name mixed type into string
df['host_name'] = df['host_name'].astype('str')

In [12]:
# Covert host_name mixed type into string
df['neighbourhood_group'] = df['neighbourhood_group'].astype('str')

In [13]:
# Covert host_name mixed type into string
df['last_review'] = df['last_review'].astype('str')

In [14]:
for col in df.columns.tolist():
  weird = (df[[col]].map(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df[weird]) > 0:
    print (col)

In [15]:
# Checking for missing values
df.isnull().sum()

id                                    0
name                                  0
host_id                               0
host_name                             0
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                           0
reviews_per_month                 49085
calculated_host_listings_count        0
availability_365                      0
number_of_reviews_ltm                 0
city                                  0
dtype: int64

In [16]:
# creat a new data frame df_nan for null values of column reviews_per_month 
df_nan = df[df['reviews_per_month'].isnull() == True]

In [17]:
df_nan.head(20)

id                                               name   host_id  \
72    219438  Super Lux View Pac Heights (3br/3.5 bath 2,600...    373342   
215   745957                               Micro-Private Single    576273   
216   747167                               Micro-Private Double    576273   
239  1145582      Charming apartment in the Fillmore, location!   6281643   
313  1273394                  Fully Furnished flat with Parking   4836128   
333  1672055  ❤️Fantastic Luxury Penthouse Apt, Panorami...    262875   
391  1087433  Masterbedrm w/ own Bath near Ucsf in Sunset 22...   5977346   
438  2048642             Luxury Modern Apartment with Bay Views   3566675   
473  1956312  Historical Elegance 2BR Castro-Duboce, CPMC Da...    413290   
499  2702606                       Murphy bed in startup office  13830783   
502  3512817                 Room in Sunny Condo - Diamond Hts.  14766201   
505  2709883                                 Rebbeca Lane SF CA  10043430   
580  2375006                   Cute room near Golden Gate Park.   3464891   
630  3907307                  Sweet spot in the Haight, central   3895812   
645  3265429             3BD Luxury SF Home 6+Month Rental Only   7345454   
691  4152054                Ocean Front Surf & Sandy Beach View  21542156   
728  5123765   Historic Marina Flat with park view, playgrounds  11269874   
785  5436115                   Private Bedroom in the Dogpatch!  18014246   
805  5560256                         Beautiful shared apartment   6334749   
828  5619003                     One Bedroom available for rent   6062018   

    host_name neighbourhood_group        neighbourhood   latitude   longitude  \
72       Phil                 nan      Pacific Heights  37.794340 -122.431670   
215    Darren                 nan   Financial District  37.794920 -122.403120   
216    Darren                 nan            Chinatown  37.794950 -122.405150   
239     Luisa                 nan     Western Addition  37.787350 -122.432250   
313    Vivian                 nan     Presidio Heights  37.787540 -122.458680   
333      Paul                 nan           Twin Peaks  37.747250 -122.445120   
391   Melissa                 nan         Outer Sunset  37.757610 -122.481170   
438     Leore                 nan      South of Market  37.770940 -122.395090   
473       Lea                 nan  Castro/Upper Market  37.765730 -122.432600   
499     David                 nan      South of Market  37.776600 -122.396740   
502       Tom                 nan      Diamond Heights  37.746113 -122.443748   
505     Nancy                 nan              Bayview  37.737990 -122.381840   
580     Geoff                 nan         Outer Sunset  37.763350 -122.495660   
630    Fakhry                 nan     Western Addition  37.770950 -122.430540   
645     Amily                 nan             Nob Hill  37.791548 -122.418445   
691     Susan                 nan       Outer Richmond  37.773140 -122.512460   
728     Derek                 nan               Marina  37.800861 -122.431139   
785    Carlos                 nan         Potrero Hill  37.759310 -122.387520   
805       Tim                 nan              Mission  37.755330 -122.414650   
828    Elaine                 nan               Marina  37.802270 -122.426910   

           room_type  price  minimum_nights  number_of_reviews last_review  \
72   Entire home/apt    649              60                  0         nan   
215     Private room    500              30                  0         nan   
216     Private room    600              30                  0         nan   
239  Entire home/apt    210              50                  0         nan   
313     Private room    500              30                  0         nan   
333  Entire home/apt    199              90                  0         nan   
391     Private room     79              30                  0         nan   
438  Entire home/apt    350              30                  0    

In [18]:
df_nan['last_review'].value_counts(dropna = False)

last_review
nan    49085
Name: count, dtype: int64

In [19]:
df_nan['number_of_reviews'].value_counts(dropna = False)

number_of_reviews
0    49085
Name: count, dtype: int64

In [20]:
df_nan['number_of_reviews_ltm'].value_counts(dropna = False)

number_of_reviews_ltm
0    49085
Name: count, dtype: int64

There are 49,085 NaN or missing values in the 'reviews_per_month' column. This coincides with missing 'last_review' data and with zero values for 'number_of_reviews' and 'number_of_reviews_ltm' in the same rows. Therefore, it can be concluded that the 49085 missing values in review_per_month columns and 0 number_of_reviews indicate that we don't have any reviews for these specific listings.

In [21]:
# Adderssing missing values
# Replace NaN values with O for reviews_per_month column
df['reviews_per_month'].fillna(0, inplace=True)

In [22]:
# Replace nan values with Unknown for neighbourhood_group column
# Check for alternative representations of missing values
print(df['neighbourhood_group'].unique())

['nan' 'Other Cities' 'Unincorporated Areas' 'City of Los Angeles'
 'Manhattan' 'Brooklyn' 'Queens' 'Bronx' 'Staten Island' 'Providence'
 'Washington' 'Newport' 'Bristol' 'Kent' 'Central Area'
 'Other neighborhoods' 'Delridge' 'Magnolia' 'Beacon Hill' 'West Seattle'
 'Ballard' 'Queen Anne' 'Capitol Hill' 'Downtown' 'Cascade'
 'Rainier Valley' 'Northgate' 'Lake City' 'University District'
 'Seward Park' 'Interbay']


In [23]:
# Replace alternative representations with NaN
df['neighbourhood_group'].replace({'nan': np.nan}, inplace=True)

In [24]:
# Replave NaN with Unknown
df['neighbourhood_group'].fillna('unknown', inplace=True)

In [25]:
# Replace nan values with Unknown for last_review column
# Check for alternative representations of missing values
# Replace alternative representations with NaN
df['last_review'].replace({'nan': np.nan}, inplace=True)

# Replave NaN with Unknown
df['last_review'].fillna(0, inplace=True)

In [26]:
df.head(20)

id                                               name  host_id  \
0      958              Bright, Modern Garden Unit - 1BR/1BTH     1169   
1     5858                                 Creative Sanctuary     8904   
2     8142  Friendly Room Apt. Style -UCSF/USF - San Franc...    21994   
3     8339                    Historic Alamo Square Victorian    24215   
4     8739                Mission Sunshine, with Private Bath     7149   
5    10537  Elegant & Cozy w/City views. Private room: Purple    36752   
6   142177                  Stylish, Spacious and Quiet Flat!   656263   
7   490931                              Mission's "La Lengua"  1269188   
8    10578                Classic Nob Hill Studio - Roof Deck    37049   
9    12041    Sunny/Sunset view room UCSF/USF - San Francisco    21994   
10  491867                  Central San Francisco Urban Dream   969518   
11   12042   Sunny/Sunset view room UCSF/USF - San Francisco*    21994   
12   12522                           Perfectly located Castro    18431   
13  144242                     Lake Street Single Family Home   699215   
14   18231                              Loft-like Apt./Garden    70224   
15  144978                  Prime Location - Small Basic Room   216682   
16  147443                 Best Views in all of San Francisco   709325   
17  493889                 Spacious 1brm w/ breathtaking view  2439917   
18   18904           Lower Haight Urban Zen With Private Bath    72464   
19  494111                Comfortable space of your own in SF   252927   

           host_name neighbourhood_group        neighbourhood  latitude  \
0              Holly             unknown     Western Addition  37.77028   
1   Philip And Tania             unknown       Bernal Heights  37.74474   
2              Aaron             unknown       Haight Ashbury  37.76555   
3               Rosy             unknown     Western Addition  37.77564   
4       Ivan & Wendy             unknown              Mission  37.76030   
5             Teresa             unknown              Bayview  37.71750   
6             Noreen             unknown       Outer Richmond  37.78440   
7            Maureen             unknown       Bernal Heights  37.74766   
8             Andrew             unknown             Nob Hill  37.79249   
9              Aaron             unknown       Haight Ashbury  37.76901   
10              Sona             unknown     Western Addition  37.77196   
11             Aaron             unknown       Haight Ashbury  37.76932   
12            Brenda             unknown  Castro/Upper Market  37.76237   
13              Kate             unknown       Inner Richmond  37.78546   
14    Arlene Helfand             unknown           Noe Valley  37.74916   
15               Jay             unknown      Pacific Heights  37.79233   
16              Mark             unknown      Diamond Heights  37.74633   
17          Germaine             unknown        Outer Mission  37.72833   
18           Patrick             unknown     Western Addition  37.77247   
19            Vivian             unknown       Outer Richmond  37.77988   

    longitude        room_type  price  minimum_nights  number_of_reviews  \
0  -122.43317  Entire home/apt    202               2                383   
1  -122.42089  Entire home/apt    235              30                111   
2  -122.45213     Private room     56              32                  9   
3  -122.43642  Entire home/apt    575               9                 28   
4  -122.42197     Private room    110               1                770   
5  -122.39698     Private room     95               1                 19   
6  -122.47932  Entire home/apt    259               5                234   
7  -122.42041     Private room    100              30                 80   
8  -122.41499  Entire home/apt     79              30                 18   
9  -122.44701     Private room     65              32                  7   
10 -122.43477  Entire home/apt    425    

In [27]:
df.tail(20)

id                                               name  \
232127  847975987116512218                   Sunny bedroom near Stanford & FB   
232128  848428736961130509                    2-BR Apartment in Western Hills   
232129  848482712915361473                       Cute 2BD Belmont Hills Oasis   
232130  848593473071213608        Ideal Location! Onsite Pool, Free Breakfast   
232131  848593734930739096     Relax and Unwind! Pets Allowed, Parking Onsite   
232132  848594164708230918      Comfort and Convenience! Kitchen, Onsite Pool   
232133  848595337740216125      Best Place to Stay in California! Onsite Pool   
232134  848595461988821158       Tour Around California! Parking, Onsite Pool   
232135  848595469433755139       Always a Great Idea To Unwind! Kitchen, Pool   
232136  848598198524262338         You Found it! Onsite Pool, Free Breakfast!   
232137  848598295563877894      Relax and Unwind! Free Breakfast, Onsite Pool   
232138  848598417306295259      Convenient Stay! Full Kitchen, Free Breakfast   
232139  848598838858396438     Enjoy a Hassle-Free Stay, Free Breakfast, Pool   
232140  848598943985944467     Best Value, Quality Stay! Pool, Free Breakfast   
232141  848599061203137315    Pets are Welcome! Outdoor Pool, Free Breakfast!   
232142  849230448624862502  Private Half Moon Bay Oasis By Pillar Point Be...   
232143  850805127216414630                                      Home near SFO   
232144  851562104616413652                   Modern New Private Studio in SFO   
232145  851792795339759410            Lux Bedroom with a Shared Bath near SFO   
232146  851801296770555568             Lux Bedroom with Private Bath near SFO   

          host_id   host_name neighbourhood_group         neighbourhood  \
232127   25821348       Wendy             unknown        East Palo Alto   
232128    2312365          Al             unknown               Belmont   
232129   43368604        Cook             unknown               Belmont   
232130  501999514   RoomPicks             unknown             San Bruno   
232131  501999514   RoomPicks             unknown             San Bruno   
232132  501999514   RoomPicks             unknown             San Bruno   
232133  501999514   RoomPicks             unknown             San Bruno   
232134  501999514   RoomPicks             unknown   South San Francisco   
232135  501999514   RoomPicks             unknown   South San Francisco   
232136  501999514   RoomPicks             unknown   South San Francisco   
232137  501999514   RoomPicks             unknown   South San Francisco   
232138  501999514   RoomPicks             unknown   South San Francisco   
232139  501999514   RoomPicks             unknown             San Bruno   
232140  501999514   RoomPicks             unknown   South San Francisco   
232141  501999514   RoomPicks             unknown             San Bruno   
232142   29395385      Khalid             unknown  Unincorporated Areas   
232143  440957696   Christina             unknown             San Bruno   
232144   41574587      Vality             unknown   South San Francisco   
232145   73133543  Stephannie             unknown   South San Francisco   
232146   73133543  Stephannie             unknown   South San Francisco   

         latitude   longitude        room_type  price  minimum_nights  \
232127  37.471170 -122.147960     Private room     80              30   
232128  37.507099 -122.295463  Entire home/apt    124               1   
232129  37.511251 -122.295454  Entire home/apt    149              14   
232130  37.640450 -122.420249  Entire home/apt    290               1   
232131  37.640140 -122.419750  Entire home/apt    290               1   
232132  37.640498 -122.418462  Entire home/apt    290               1   
232133  37.640306 -122.418981  Entire home/apt    323               1   
232134  37.642191 -122.419528  Entire home/apt    323               1   
232135  37.642353 -122.419012  Entire home/apt    323               1   
232136  37.64241

In [28]:
df.isnull().sum()

id                                0
name                              0
host_id                           0
host_name                         0
neighbourhood_group               0
neighbourhood                     0
latitude                          0
longitude                         0
room_type                         0
price                             0
minimum_nights                    0
number_of_reviews                 0
last_review                       0
reviews_per_month                 0
calculated_host_listings_count    0
availability_365                  0
number_of_reviews_ltm             0
city                              0
dtype: int64

In [29]:
# Checking for Duplicate Values 
df_dups = df[df.duplicated()]

In [30]:
df_dups

Empty DataFrame
Columns: [id, name, host_id, host_name, neighbourhood_group, neighbourhood, latitude, longitude, room_type, price, minimum_nights, number_of_reviews, last_review, reviews_per_month, calculated_host_listings_count, availability_365, number_of_reviews_ltm, city]
Index: []

No Duplicates found

In [31]:
df.shape

(232147, 18)

# 5. Adjusting variable names for better understanding

In [32]:
# Change "id" column name to "listing_id".
df.rename(columns = {'id' : 'listing_id'}, inplace = True)

In [33]:
# Change "name" column name to "listing_name".
df.rename(columns = {'name' : 'listing_name'}, inplace = True)

In [34]:
# Change "neighbourhood" column name to "neighborhood".
df.rename(columns = {'neighbourhood' : 'neighborhood'}, inplace = True)

In [35]:
# Change "number_of_reviews_ltm" column name to "last_12_month_reviews_count".
df.rename(columns = {'number_of_reviews_ltm' : 'last_12_month_reviews_count'}, inplace = True)

# 6. Removing irrelevant columns

In [36]:
# Drop "neighbourhood_group" columns in df (has too many unknowns / information is in "neighborhood" column already)
df = df.drop(columns = ['neighbourhood_group'])

In [37]:
df.head(20)

listing_id                                       listing_name  host_id  \
0          958              Bright, Modern Garden Unit - 1BR/1BTH     1169   
1         5858                                 Creative Sanctuary     8904   
2         8142  Friendly Room Apt. Style -UCSF/USF - San Franc...    21994   
3         8339                    Historic Alamo Square Victorian    24215   
4         8739                Mission Sunshine, with Private Bath     7149   
5        10537  Elegant & Cozy w/City views. Private room: Purple    36752   
6       142177                  Stylish, Spacious and Quiet Flat!   656263   
7       490931                              Mission's "La Lengua"  1269188   
8        10578                Classic Nob Hill Studio - Roof Deck    37049   
9        12041    Sunny/Sunset view room UCSF/USF - San Francisco    21994   
10      491867                  Central San Francisco Urban Dream   969518   
11       12042   Sunny/Sunset view room UCSF/USF - San Francisco*    21994   
12       12522                           Perfectly located Castro    18431   
13      144242                     Lake Street Single Family Home   699215   
14       18231                              Loft-like Apt./Garden    70224   
15      144978                  Prime Location - Small Basic Room   216682   
16      147443                 Best Views in all of San Francisco   709325   
17      493889                 Spacious 1brm w/ breathtaking view  2439917   
18       18904           Lower Haight Urban Zen With Private Bath    72464   
19      494111                Comfortable space of your own in SF   252927   

           host_name         neighborhood  latitude  longitude  \
0              Holly     Western Addition  37.77028 -122.43317   
1   Philip And Tania       Bernal Heights  37.74474 -122.42089   
2              Aaron       Haight Ashbury  37.76555 -122.45213   
3               Rosy     Western Addition  37.77564 -122.43642   
4       Ivan & Wendy              Mission  37.76030 -122.42197   
5             Teresa              Bayview  37.71750 -122.39698   
6             Noreen       Outer Richmond  37.78440 -122.47932   
7            Maureen       Bernal Heights  37.74766 -122.42041   
8             Andrew             Nob Hill  37.79249 -122.41499   
9              Aaron       Haight Ashbury  37.76901 -122.44701   
10              Sona     Western Addition  37.77196 -122.43477   
11             Aaron       Haight Ashbury  37.76932 -122.44723   
12            Brenda  Castro/Upper Market  37.76237 -122.42992   
13              Kate       Inner Richmond  37.78546 -122.46148   
14    Arlene Helfand           Noe Valley  37.74916 -122.42971   
15               Jay      Pacific Heights  37.79233 -122.42431   
16              Mark      Diamond Heights  37.74633 -122.44591   
17          Germaine        Outer Mission  37.72833 -122.44125   
18           Patrick     Western Addition  37.77247 -122.43301   
19            Vivian       Outer Richmond  37.77988 -122.48488   

          room_type  price  minimum_nights  number_of_reviews last_review  \
0   Entire home/apt    202               2                383     2/19/23   
1   Entire home/apt    235              30                111      8/6/17   
2      Private room     56              32                  9    10/27/22   
3   Entire home/apt    575               9                 28     6/28/19   
4      Private room    110               1                770     2/25/23   
5      Private room     95               1                 19     1/28/23   
6   Entire home/apt    259               5                234     2/17/23   
7      Private room    100              30                 80    11/26/22   
8   Entire home/apt     79              30                 18     5/17/15   
9      Private room     65              32                  7     6/11/22   
10  Entire home/apt    425               2                 56     2/19/23   
11     Private room     65              32            

In [38]:
df.shape

(232147, 17)

# 7. Exporting Data

In [39]:
# Export df_ords as orders_checked
df.to_csv(os.path.join(path,'02 Data','Prepared Data', 'AB_US_2023_cleaned.csv'))